src/infrastructure/idAuthentication/DokobitAuthenticationProvider.ts

```diff
@@ -45,8 +49,29 @@ export class DokobitAuthenticationProvider implements AuthenticationProvider {
 
     return {
       code: sessionStatus.code,
-      firstName: sessionStatus.name,
-      lastName: sessionStatus.surname,
+      profile: mapToProfile(sessionStatus),
```

Comment:
Cool, wasn't even thinking about doing it on this level, but it's cleaner indeed.